app/models/segment.rb

```diff
@@ -32,9 +32,11 @@ def self.with_ends
   # Returns the Segment the runner would be on at the given run time. A run time is a Duration since the start of the
   # run.
   #
-  # If the Duration takes place outside the run (i.e. is greater than Run#duration), nil is returned.
+  # If the Duration takes place outside the run (i.e. is greater than Run#duration), the last segment is returned. This
+  # is to support blind runs where the number of segments is not known, and new segments can be added on to the end of
+  # the run as they are reached.
   def self.at_run_time(run_time)
-    with_ends.order(segment_number: :asc).where('realtime_end_ms > ?', run_time.to_ms).first
+    with_ends.order(segment_number: :asc).where('realtime_end_ms > ?', run_time.to_ms).first || last
```

Comment:
Took this comment for  me to realize we're actually not using this function anywhere anymore, I think since we replaced the "Current split" column of the race entries table with a full timeline. I think I made this change so that runners who were finished would still show their final split rather than nothing, before replacing the whole thing with a timeline.

I'll undo the change, might keep the function around though just because it could come in handy for auto-coaching.